### PR TITLE
migrations and functionality for csv

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

File: polls/admin.py [hunk 1]
```diff
@@ -30,7 +33,34 @@ def queryset(self, request, queryset):
...
+    field_names = ['title', 'date_submitted', 'user', 'answer']
+
+    writer.writerow(field_names)
+
+    for question in queryset:
+        if hasattr(question, 'choicevote_set'):
+            for choice in question.choicevote_set.all():
+                writer.writerow([
+                    choice.question, choice.submission_date, choice.user,
+                    choice.user, choice.answer])
```
Comment (miltontony) [inline polls/admin.py:53]: duplicate field `choice.user`

File: polls/admin.py [hunk 2]
```diff
@@ -30,7 +33,34 @@ def queryset(self, request, queryset):
...
+
+    field_names = ['title', 'date_submitted', 'user', 'answer']
+
+    writer.writerow(field_names)
+
+    for question in queryset:
+        if hasattr(question, 'choicevote_set'):
+            for choice in question.choicevote_set.all():
+                writer.writerow([
+                    choice.question, choice.submission_date, choice.user,
```
Comment (miltontony) [inline polls/admin.py]: consider using `question.title` instead of `choice.question` - more explicit

File: polls/admin.py [hunk 3]
```diff
@@ -30,7 +33,34 @@ def queryset(self, request, queryset):
...
+                    choice.user, choice.answer])
+
+        elif hasattr(question, 'freetextvote_set'):
+            for choice in question.freetextvote_set.all():
+                writer.writerow([
+                    choice.question, choice.submission_date, choice.user,
+                    choice.user, choice.answer])
```
Comment (miltontony) [inline polls/admin.py:59]: duplicate field `choice.user`

**Comment by codiebeulaine** [comment]
@miltontony @justinvdm ready for review

File: polls/admin.py [hunk 4]
```diff
@@ -30,7 +33,34 @@ def queryset(self, request, queryset):
         return queryset
 
 
+def download_as_csv(QuestionAdmin, request, queryset):
```
Comment (justinvdm) [inline polls/admin.py]: Could we lowercase `QuestionAdmin` since it is an instance? Sorry, not-so-useful bikeshed.

File: polls/admin.py [hunk 5]
```diff
@@ -30,7 +33,34 @@ def queryset(self, request, queryset):
...
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment;filename=export.csv'
```
Comment (justinvdm) [inline polls/admin.py]: Could we make the filename more descriptive? When someone has multiple downloads, they would get confused about which export is which.

Maybe it should contain the export date and the fact that it is a question export? Something like `tuneme-questions-2016-01-11.csv` or something?

File: polls/tests/test_admin.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        response = download_as_csv(QuestionAdmin,
+                                   None,
+                                   Question.objects.all())
+        date = str(datetime.datetime.now().date())
+        expected_output = ('Content-Type: text/csv\r\nContent-Disposition:'
+                           ' attachment;filename=export.csv\r\n\r\n'
+                           'title,date_submitted,user,answer'
+                           '\r\nis this a test,'
+                           + date + ',tester,'
+                           'this is an answer\r\n')
```
Comment (justinvdm) [inline polls/tests/test_admin.py:123]: I think we'll need to quote the csv fields to handle cases where a field value has `,`s in it.

Comment (codiebeulaine) [inline polls/tests/test_admin.py:123]: @justinvdm the csv writer does that automatically

Comment (justinvdm) [inline polls/tests/test_admin.py:123]: Ah, I see, thanks.

File: polls/tests/test_admin.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        # make a vote
+        client = Client()
+        client.login(username='tester', password='tester')
+
+        client.post(reverse('molo.polls:vote',
+                    kwargs={'question_id': question.id}),
+                    {'choice': [choice1.id, choice2.id]})
+        # should automatically create the poll vote
+        # test poll vote
+        response = download_as_csv(QuestionAdmin,
```
Comment (justinvdm) [inline polls/tests/test_admin.py]: Should we pass in a `QuestionAdmin` instance to match what would actually be passed in? Similarly for the other test.

**Comment by justinvdm** [comment]
Left a few comments, looks good.

**Comment by codiebeulaine** [comment]
@justinvdm changed according to comments, ready for review :)

**Comment by justinvdm** [comment]
@codiebeulaine seeing an [unresolved thing](https://github.com/praekelt/molo-tuneme/pull/145#discussion_r49315209).

**Comment by justinvdm** [comment]
:+1: when travis is happy

**Pull request merged**